pkg/mattermost/database_external.go

```diff
@@ -0,0 +1,115 @@
+package mattermost
+
+import (
+	"fmt"
+
+	mattermostv1beta1 "github.com/mattermost/mattermost-operator/apis/mattermost/v1beta1"
+	"github.com/mattermost/mattermost-operator/pkg/database"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type ExternalDBConfig struct {
+	secretName         string
+	dbType             string
+	hasReaderEndpoints bool
+	hasDBCheckURL      bool
+}
+
+func NewExternalDBConfig(mattermost *mattermostv1beta1.Mattermost, secret corev1.Secret) (*ExternalDBConfig, error) {
+	if mattermost.Spec.Database.External == nil {
+		return nil, fmt.Errorf("external database config not provided")
```

Comment:
```suggestion
		return nil, errors.New("external database config not provided")
```

There are many more of these below.